src/aiida/tools/archive/imports.py

```diff
@@ -262,11 +262,18 @@ def _import_users(
     # get matching emails from the backend
     output_email_id: Dict[str, int] = {}
     if input_id_email:
-        output_email_id = dict(
-            orm.QueryBuilder(backend=backend_to)
-            .append(orm.User, filters={'email': {'in': list(input_id_email.values())}}, project=['email', 'id'])
-            .all(batch_size=query_params.batch_size)
-        )
+        output_email_id = {
+            key: value
+            for query_results in [
+                dict(
+                    orm.QueryBuilder(backend=backend_to)
+                    .append(orm.User, filters={'email': {'in': chunk}}, project=['email', 'id'])
+                    .all(batch_size=query_params.batch_size)
+                )
+                for _, chunk in batch_iter(set(input_id_email.values()), query_params.filter_size)
```

Comment:
Note that I use `set` to only get a set of unique `uuids` so that the query will not contain duplicates. I don't think this will ever happen but I wanted to be better safe here. Also I guess the subsequent code can also deal with duplicates.